UnityProject/Assets/Scripts/Mobs/AI/MobPathfinderV2.cs

```diff
@@ -0,0 +1,444 @@
+		public virtual void OnDisable()
+		{
+			if (CustomNetworkManager.IsServer)
+			{
+				Movement.OnLocalTileReached.RemoveListener(OnTileReached);
+			}
+		}
+
+		void OnDrawGizmos()
+		{
+			if (debugPath != null)
+			{
+				for (var i = 1; i < debugPath.Count; ++i)
+				{
+					var from = new Vector3(debugPath[i - 1].position.x + 1, debugPath[i - 1].position.y + 1, 0);
+					var to = new Vector3(debugPath[i].position.x + 1, debugPath[i].position.y + 1, 0);
+					Gizmos.DrawLine(from, to);
+				}
+			}
+
+			if (debugSearch.Count > 0)
+			{
+				foreach (KeyValuePair<Vector2, Color> kvp in debugSearch)
+				{
+					Gizmos.color = kvp.Value;
+					Gizmos.DrawCube(transform.parent.TransformPoint((Vector3)kvp.Key), Vector3.one * 0.9f);
+				}
+			}
+		}
+
+		///<summary>
+		/// Use local positions related to the matrix the object is on
+		///</summary>
+		public List<Node> FindNewPath(Vector2Int startPos, Vector2Int targetPos)
+		{
+			if (startPos == targetPos) return null;
+
+			pathFound = false;
+
+			startNode = new Node
+			{
+				position = startPos,
+				distanceTraveled = 0
+			};
+
+			goalNode = new Node
+			{
+				position = targetPos
+			};
+
+			if (Vector2Int.Distance(startNode.position, goalNode.position) < 2f)
+			{
+				//This is just the next tile over, create a path with start and goal nodes
+				return new List<Node>(new Node[] { startNode, goalNode });
+			}
+
+			isComplete = false;
+			isComplete = false;
+			frontierNodes = new PriorityQueue<Node>();
+			frontierNodes.Enqueue(startNode);
+			exploredNodes.Clear();
+			allNodes.Clear();
+			allNodes.Add(startNode.position, startNode);
+			allNodes.Add(goalNode.position, goalNode);
+			return SearchForRoute();
+		}
+
+		List<Node> SearchForRoute()
+		{
+			int timeOut = 0;
+
+			while (!isComplete)
+			{
+				timeOut++;
+				if (timeOut > 200)
+				{
+					isComplete = true;
+					//This could be because you are trying to use a goal node that is inside a wall or the path was blocked
+					Logger.Log(
+						$"Pathing finding could not find a path where one was expected to be found. StartNode {startNode.position} GoalNode {goalNode.position}",
+						Category.Movement);
+					return null;
+				}
+
+				if (frontierNodes.Count > 0)
+				{
+					Node currentNode = frontierNodes.Dequeue();
+
+					if (currentNode.neighbors == null)
+					{
+						FindNeighbours(currentNode);
+					}
+
+					if (!exploredNodes.Contains(currentNode))
+					{
+						exploredNodes.Add(currentNode);
+					}
+
+					ExpandFrontierAStar(currentNode);
+
+					if (pathFound)
+					{
+						isComplete = true;
+						List<Node> path = new List<Node>();
+
+						Node nextNode = goalNode;
+
+						do
+						{
+							path.Add(nextNode);
+							nextNode = nextNode.previous;
+						} while (nextNode != null);
+
+						path.Reverse();
+
+						return path;
+					}
+				}
+				else
+				{
+					isComplete = true;
+					return null;
+				}
+			}
+
+			return null;
+		}
+
+		void FindNeighbours(Node currentNode)
```

Comment:
You could use the meta data system Neighbours Use by atmospherics

Reply:
I'm planning on updating the pathfinder in later PRs, but for now it uses the already existing code

Reply:
as in copy and paste from the old pathfinder?

Reply:
for now it is, yes.
it only has some elements updated to use things like the new UOP stuff